src/libraries/System.Private.CoreLib/src/System/IO/Path.cs

```diff
@@ -654,83 +654,139 @@ private static unsafe string JoinInternal(ReadOnlySpan<char> first, ReadOnlySpan
                 return string.Create(
```

Comment:
This one should be smaller and faster as:
```
return hasSeparator ? string.Concat(first, second) : string.Concat(first, PathInternal.DirectorySeparatorCharAsString, second);
```
(The existing implementation may need to stay for `MS_IO_REDIST`.)

Reply:
Good suggestion!